addons/account/models/account_bank_statement.py

```diff
@@ -341,7 +341,14 @@ def button_reopen(self):
 
     def action_bank_reconcile_bank_statements(self):
         self.ensure_one()
-        bank_stmt_lines = self.mapped('line_ids')
+        limit = int(self.env["ir.config_parameter"].get_param("account.reconcile.batch", 1000))
+        bank_stmt_lines = self.env['account.bank.statement.line'].search([
+            ('statement_id', 'in', self.ids),
+            # take not reconciled lines only. See _check_lines_reconciled method
+            ('account_id', '=', False),
+            ('journal_entry_ids', '=', False),
+            ('amount', '!=', 0),
+        ], limit=limit)
```

Comment:
Don't force a `search` since everything is probably already in cache so `self.line_ids.filtered(lambda line: ...)`

Reply:
That cache is always empty, until ``action_bank_reconcile_bank_statements`` is called by another method, which is not the case.

You can visit rd-performance channel in discord to discuss it :relaxed:

Reply:
If you call the method from a RPC call, yes. Suppose you override the method with a super call:
- your search is made by the super call.
- another search is made by the ORM for the same result.
=> Let the ORM do its job :)

Reply:
There is no known module that overrides this method (including in OCA).
Even if somebody has such module, I'd prefer such users make search request twice than let all other users suffer from slow filtering thousands of lines in python

Reply:
Ok. You got the point.